src/platform/datapath_raw_xdp_win.c

```diff
@@ -371,6 +382,38 @@ CxPlatDpRawInterfaceUninitialize(
     #pragma warning(pop)
 }
 
+QUIC_STATUS
+GetTxOffloadConfig(
+    _In_ CXPLAT_QUEUE* Queue,
+    _Out_ uint32_t* TxChecksumOffload
+    )
+{
+    QUIC_STATUS Status;
+    XDP_CHECKSUM_CONFIGURATION ChecksumConfig;
+    uint32_t OptionLength = sizeof(ChecksumConfig);
+
+    Status =
+        XskGetSockopt(
+            Queue->TxXsk, XSK_SOCKOPT_TX_OFFLOAD_CURRENT_CONFIG_CHECKSUM, &ChecksumConfig,
+            &OptionLength);
+    if (QUIC_FAILED(Status) ||
+        OptionLength < XDP_SIZEOF_CHECKSUM_CONFIGURATION_REVISION_1 ||
+        ChecksumConfig.Header.Revision != XDP_CHECKSUM_CONFIGURATION_REVISION_1 ||
+        ChecksumConfig.Header.Size < XDP_SIZEOF_CHECKSUM_CONFIGURATION_REVISION_1) {
```

Comment:
So, if XDP starts returning a version 2, it breaks our support for the offload? If so, that doesn't seem like a great design.

Reply:
Version 2 is by definition a breaking change, and version 2 has not yet been defined. The only legal action the application can take here is to ignore the payload.

Reply:
I don't have a strong opinion, other than that XDP already consistently uses this pattern everywhere, and I don't see much justification to deviate here.